src/swish/io.ss

```diff
@@ -689,32 +689,44 @@
       (define (gp) fp)
       (define (sp! pos) (set! fp pos))
       (define (close) (close-osi-port port))
-      (case type
-        [(binary-input)
-         (make-custom-binary-input-port name r! gp sp! close)]
-        [(binary-output)
-         (make-custom-binary-output-port name w! gp sp! close)]
-        [(input)
-         (binary->utf8
-          (make-custom-binary-input-port name r! gp sp! close))]
-        [(output)
-         (binary->utf8
-          (make-custom-binary-output-port name w! gp sp! close))]
-        [(append)
-         (binary->utf8
-          (make-custom-binary-output-port name a! #f #f close))])))
-
-  (define (open-file-to-read name)
-    (open-file name O_RDONLY 0 'input))
-
-  (define (open-file-to-write name)
-    (open-file name (+ O_WRONLY O_CREAT O_EXCL) #o777 'output))
-
-  (define (open-file-to-append name)
-    (open-file name (+ O_WRONLY O_CREAT O_APPEND) #o777 'append))
-
-  (define (open-file-to-replace name)
-    (open-file name (+ O_WRONLY O_CREAT O_TRUNC) #o777 'output))
+      (let open ([type type])
+        (case type
+          [(binary-input)
+           (make-custom-binary-input-port name r! gp sp! close)]
+          [(binary-output)
+           (make-custom-binary-output-port name w! gp sp! close)]
+          [(binary-append)
+           (make-custom-binary-output-port name a! #f #f close)]
+          [(input) (binary->utf8 (open 'binary-input))]
+          [(output) (binary->utf8 (open 'binary-output))]
+          [(append) (binary->utf8 (open 'binary-append))]))))
+
+  (define open-file-to-read
+    (case-lambda
+     [(name) (open-file-to-read name #f)]
+     [(name binary?)
+      (open-file name O_RDONLY 0 (if binary? 'binary-input 'input))]))
+
+  (define open-file-to-write
+    (case-lambda
+     [(name) (open-file-to-write name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_EXCL) #o777
+        (if binary? 'binary-output 'output))]))
+
+  (define open-file-to-append
+    (case-lambda
+     [(name) (open-file-to-append name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_APPEND) #o777
+        (if binary? 'binary-append 'append))]))
+
+  (define open-file-to-replace
+    (case-lambda
+     [(name) (open-file-to-replace name #f)]
+     [(name binary?)
+      (open-file name (+ O_WRONLY O_CREAT O_TRUNC) #o777
```

Comment:
Use #o666 to be consistent with Chez Scheme. On Windows, it's ignored.

Reply:
Good idea.